runtime/ms_rest_azure/spec/azure_service_client_spec.rb

```diff
@@ -37,7 +37,7 @@ module MsRestAzure
       azure_service_client.long_running_operation_retry_timeout = 0
 
       allow_any_instance_of(MsRestAzure::PollingState).to receive(:create_connection).and_return(nil)
-      allow(azure_service_client).to receive(:update_state_from_azure_async_operation_header) do |request, polling_state|
+      allow(azure_service_client).to receive(:update_state_without_custom_deserialization) do |request, polling_state|
```

Comment:
with this change, how are we validating the intent of the test "should use async operation header when async_operation_header present" ? it looks like we're using the same method name in both sets of tests (for async operation header and for location header check.